assets/javascripts/discourse/initializers/filter-category-list-for-anon.js

```diff
@@ -0,0 +1,31 @@
+import CategoryList from "discourse/models/category-list";
+import PreloadStore from "discourse/lib/preload-store";
+import { ajax } from "discourse/lib/ajax";
+
+export default {
+  name: "filter-category-list-for-anon",
+
+  initialize(container) {
+    const siteSettings = container.lookup("service:site-settings");
+    if (siteSettings.discourse_global_filter_enabled) {
+      CategoryList.reopenClass({
+        list(store) {
+          const tagParam = new URLSearchParams(window.location.search).get(
+            "tag"
+          );
+          const getCategories = () => ajax(`/categories.json?tag=${tagParam}`);
```

Comment:
```suggestion
          // this tag parameter needs to be passed (core doesn't do this by default)
          const getCategories = () => ajax(`/categories.json?tag=${tagParam}`);
```

A comment would be nice here, feel free to reword.